pkg/registries/registries.go

```diff
@@ -140,14 +140,20 @@ func mergeMirrors(existingMirrors, newMirrors []mirror) []mirror {
 }
 
 // generateConfig is a serialization method that generates a valid TOML representation from a mirrorSet object.
-// Results in content usable as a containerd image registry configuration file.
+// Results in content usable as a containerd image registry configuration file. Returns empty string if no mirrors exist
 func (ms *mirrorSet) generateConfig() string {
 	result := ""
+	if len(ms.mirrors) == 0 {
+		return result
+	}
 
-	if ms.mirrorSourcePolicy == config.AllowContactingSource {
-		result += fmt.Sprintf("server = \"https://%s\"", ms.source)
-		result += "\r\n\r\n"
+	fallbackServer := ms.source
+	if ms.mirrorSourcePolicy == config.NeverContactSource {
+		// set the fallback server to the first mirror to ensure the source is never contacted, even if all mirrors fail
+		fallbackServer = ms.mirrors[0].host
 	}
+	result += fmt.Sprintf("server = \"https://%s\"", fallbackServer)
+	result += "\r\n\r\n"
```

Comment:
This is getting a bit confusing, please add some documentation to this clarifying the expected input/output.

Reply:
added some comments. LMK if the unit tests are sufficient for expected in/output